ui/apps/platform/src/services/ReportsService.types.ts

```diff
@@ -110,22 +110,35 @@ export type ReportHistoryResponse = {
     reportSnapshots: ReportSnapshot[];
 };
 
-export type ReportSnapshot = Snapshot & {
+export type ViewBasedReportSnapshot = Snapshot & {
+    viewBasedVulnReportFilters: ViewBasedVulnerabilityReportFilters;
+    requestName: string;
+    areaOfConcern: string;
+};
+
+export type ConfiguredReportSnapshot = Snapshot & {
     reportConfigId: string;
     vulnReportFilters: VulnerabilityReportFilters;
     collectionSnapshot: CollectionSnapshot;
     schedule: Schedule | null;
     notifiers: NotifierConfiguration[];
 };
 
-// @TODO: Technically, this type will have the same fields as ReportSnapshot but the irrelevant
-// ones will be null or empty. For now, I didn't include them
-export type ViewBasedReportSnapshot = Snapshot & {
-    requestName: string;
-    isViewBased: boolean;
-    areaOfConcern: string;
-    vulnReportFilters: ViewBasedVulnerabilityReportFilters;
-};
+export type ReportSnapshot = ConfiguredReportSnapshot | ViewBasedReportSnapshot;
```

Comment:
@ksurabhi91 @charmik-redhat We try to align our types closely to how the protos are set up. I know we're reusing the same object `ReportSnapshot` for both view-based report snapshots and the already existing report snapshots. In the UI, I've made a distinction between both types here. This helps TypeScript understand the differences in code when accessing certain fields. Would this sort of terminology translate well back to the backend protos for any future refactoring?